Fastsim digirecomix fixes

lveldere opened the pull request:
a few fixes were made wrt #7306 
- careful comparison of fastsim cfg dumps before and after changes revealed a few cfg mistakes which are fixed
- fastsim unit tests were fixed

Commit:
initial commit

Commit:
merged

Commit:
more

Commit:
- disable track mixing in case of gen-level mixing
- correct label for pileup information in case of gen-level mixing

Commit:
complete the event content for fastsim pu files

Commit:
fastsim: digi-reco mixing: fixes after detailed review of effects on dumped fastsim cfgs

Commit:
fastsim: fix unit tests after transition to digi-reco mixing as default (unit tests stay on gen-mixing for the moment)

cmsbuild commented:
A new Pull Request was created by @lveldere  for CMSSW_7_4_X.

Fastsim digirecomix fixes

It involves the following packages:

Configuration/Applications
Configuration/PyReleaseValidation
Configuration/StandardSequences
FastSimulation/Configuration
FastSimulation/PileUpProducer


@civanch, @nclopezo, @lveldere, @boudoul, @franzoni, @mdhildreth, @cmsbuild, @srimanob, @davidlange6 can you please review it and eventually sign? Thanks.
@ghellwig, @Martin-Grunewald, @makortel, @GiacomoSguazzoni, @rovere, @VinInn, @appeltel, @matt-komm, @cerati, @dgulhan this is something you requested to watch as well.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.

cmsbuild commented:
+1
Tested at: 464dcc51b47d3b635d20d52c4778033f69202ccc
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7358/2106/summary.html

Commit:
- recovered option --pileup default for cmsDriver
- default input files for gen-mixing are now 8 TeV
- when postLS1 customisation is used, input files for gen-mixing are 13 TeV
- HLT step runs independent from RECO step (broken during transition to digi-reco mixing)

cmsbuild commented:
Pull request #7358 was updated. @civanch, @nclopezo, @lveldere, @boudoul, @franzoni, @mdhildreth, @cmsbuild, @srimanob, @davidlange6 can you please check and sign again.

lveldere commented:
@cmsbuild 

I made a new commit,
could you retrigger tests?
thanks
Lukas

lveldere commented:
list of tests performed

- all FastSim PU scenarios in runTheMatrix run
   (tested wfs 5.1,135.1,400.0,25400.0)

- impact of this branch on FastSim cfgs was studied in detail
   - python cfg files were dumped for all listed wfs in 740pre5 and in this branch
   - for each dumped cfg the differences between 740pre5 and this branch were listed and understood
   - a similar comparison was done for a variation of wf 135.1 re-configured for digi-reco mixing

- all fastsim unit tests run

- the HLT step is shown to run independent from the RECO step

davidlange6 commented:
remaining changes indeed are fastsim fixes for issues in PR - I will bypass for pre6. Please comment if there is something I missed in this PR.
